railties/test/application/test_runner_test.rb

```diff
@@ -576,6 +576,34 @@ def test_warnings
         capture(:stderr) { run_test_command("test/models/warnings_test.rb -w") })
     end
 
+    def test_reset_sessions_before_rollback
```

Comment:
Can you change this to be `test_reset_sessions_before_rollback_on_system_tests`? I think that's a better example of what this test is testing since it only applies to system tests.